### ISSUE TITLE
Support for pipeline events

### ISSUE THREAD

**Issue by yegorski** [nazrhyn/gitlab-slack#27]
Do you plan to listen for "Pipeline events"?

One thing I can see being useful is custom notifications on important events, such as `master` build failing, etc, where the Slack integration would do an `@here` to the channel.

**Comment by nazrhyn** [comment]
Actually, we're about to dig into GitLab CI soon, at work. I will definitely need to make some modifications for that, and will also fix some more of these pending bugs and vulnerable packages.

**Comment by nazrhyn** [comment]
I was able to reorganize my 10% time at work :tada:, which will give me more time to work on this, periodically. I just released [2.1.0](https://github.com/nazrhyn/gitlab-slack/releases/tag/2.1.0) which resolved 5 of the pending issues and brings the package dependencies and the code up to date.

Pipeline events are next!

**Comment by nazrhyn** [comment]
@genru I noticed that you have a fork with some work done for the pipeline and build events. When I go to implement those here, may I use your code either directly or as a reference?

**Comment by genru** [comment]
@nazrhyn sure, please. but have no any guarantee

**Comment by genru** [comment]
need i make a pull request?

**Comment by nazrhyn** [comment]
@genru Whatever you prefer. I am happy to simply use it as reference, especially if you think there may be improvements to be made.

**Comment by genru** [comment]
@nazrhyn  https://github.com/nazrhyn/gitlab-slack/pull/29

**Comment by nazrhyn** [comment]
@genru @yegorski I'm starting this and was interested to hear your thoughts about something:

When would you want to see notifications for individual jobs/builds within a pipeline? Wouldn't that be potentially pretty spammy?

If a pipeline fails, then the builds list can be looked at to determine which job/build has failed, and then that can be sent to Slack, but actually showing every single job/build seems like overkill.

**Comment by genru** [comment]
@nazrhyn Yes, pipeline hooks everything. it is a bit too many events. I thought only notify key pipeline events would be nice.

I dont understand what do you mean "the builds list can be looked at to determine which job/build has failed". I use `.gitlab-ci.yaml` do my CI work, In my scenario, when a job failed just has one message `pipeline xxxx by xxx failed` comes in.

**Comment by nazrhyn** [comment]
Your first response is kind of what I was looking for. We'll see how it goes!
